tests/test.py

```diff
@@ -0,0 +1,185 @@
+        self.success = True
+
+    def send_loop(self):
+        for i in range(0, 10):
+            self.mav.mav.ping_send(
+                int(time.time() * 1000), i, self.targetSysId,
+                self.targetCompId)
+            sleep(0.5)
+
+    def run(self):
+        self.sender_thread.start()
+
+        while (True):
+            msg = self.mav.recv_match(blocking=True, timeout=5)
+            if msg is not None:
+                if self.targetSysId != 0 and msg.get_srcSystem(
+                ) != self.targetSysId:
+                    log("Received unexpected response from %d/%d - current target: %d/%d"
+                        % (msg.get_srcSystem(), msg.get_srcComponent(),
+                           self.targetSysId, self.targetCompId))
+                    self.success = False
+
+                # TODO maybe check if no message sent has seq bigger than any sent?
+                self.received.append(msg)
+            else:
+                break
+
+    def accept(self, func):
+        return self.success and func(self.received)
+
+
+class MavlinkReceiver(Thread):
+    '''MAVLink message receiver'''
+
+    def __init__(self, name, port, sysId, compId):
+        super().__init__()
+        self.name = name
+        self.sysId = sysId
+        self.compId = compId
+        self.mav = mavutil.mavlink_connection(
+            'udpin:127.0.0.1:' + str(port), source_system=self.sysId)
+        self.received = []
+        self.success = True
+
+    def run(self):
+        while (True):
+            msg = self.mav.recv_match(blocking=True, timeout=5)
+            if msg is not None:
+                self.received.append(msg)
+                if msg.target_system not in [0, self.sysId]:
+                    log("Received unexpected message from: %d/%d, meant to %d/%d - current receiver: %d/%d"
+                        % (msg.get_srcSystem(), msg.get_srcComponent(),
+                           msg.target_system, msg.target_component, self.sysId,
+                           self.compId))
+                    self.success = False
+                self.mav.mav.ping_send(
+                    int(time.time() * 1000), msg.seq,
+                    msg.get_srcSystem(), msg.get_srcComponent())
+            else:
+                break
+
+    def accept(self, func):
+        return self.success and func(self.received)
+
+
+# Setup mavlink-router
+proc = subprocess.Popen(
+    [
+        sys.argv[1], "-e", "127.0.0.1:10100", "-e", "127.0.0.1:10101",
+        "127.0.0.1:10000"
+    ],
+    stderr=log_file.fileno(),
+    stdout=log_file.fileno())
+
+# Two senders: one send to all (target 0). The other sends to target 100
+sender0 = MavlinkSender("sender0", 10000, 1, 1, 0, 0)
+sender100 = MavlinkSender("sender100", 10000, 2, 1, 100, 1)
+
+# Two receivers.
+receiver100 = MavlinkReceiver("receiver100", 10100, 100, 0)
+receiver101 = MavlinkReceiver("receiver101", 10101, 101, 0)
+
+sender100.start()
+# Wait a bit before starting broadcaster pinger
```

Comment:
You could instead make everybody send heartbeat and wait for that instead arbitrary delay.